File: sample/src/main/java/com/readdle/swiftjava/sample/SwiftEnvironment.java
package com.readdle.swiftjava.sample;

import com.readdle.codegen.anotation.SwiftReference;

@SwiftReference
public class SwiftEnvironment {

    // Swift SwiftEnvironment private native pointer
    private long nativePointer = 0L;

    // Swift SwiftEnvironment private constructor
    // Should be private. Don't call this constructor from Java!
    private SwiftEnvironment() {}

    // Swift SwiftEnvironment release method
    private native void release();

    public static native void initEnvironment();

}

File: sample/src/main/java/com/readdle/swiftjava/sample/SwiftEnvironment.kt
package com.readdle.swiftjava.sample

import com.readdle.codegen.anotation.SwiftReference

// Swift JNI private constructor
// Should be private. Don't call this constructor from Java!
@SwiftReference
class SwiftEnvironment private constructor() {

    // Swift JNI private native pointer
    private val nativePointer = 0L

    // Swift JNI release method
    external fun release()

    companion object {

        @JvmStatic
        external fun initEnvironment()

        @JvmStatic
        external fun is64BitArch(): Boolean
    }

}
